spec/models/green_flag/feature_spec.rb

```diff
@@ -1,135 +1,187 @@
 
-      context 'when there are no decisions' do
-        subject { feature.forget_non_manual_decisions!(enabled) }
+    subject { feature.latest_version }
 
-        it 'should not attempt to delete non-manual decisions' do
-          expect(GreenFlag::FeatureDecision.non_manual).to receive(:delete_all).never
+    context 'when there are rules' do
+      before(:each) do
+        @latest_version_number = 15
+
+        (0...3).each do |i|
+          FactoryGirl.create(:green_flag_rule, feature_id: feature.id, version_number: @latest_version_number - i) 
         end
+      end
 
-        it 'should not create a feature event' do
-         expect(GreenFlag::FeatureEvent.count).to eq 0
-       end
-     end
-
-     context 'when there are manual decisions' do
-       subject { feature.forget_non_manual_decisions!(enabled) }
-
-       before(:each) do
-         @manual_fd = GreenFlag::FeatureDecision.create!(feature_id: feature.id, manual: true, site_visitor_id: 1)
-       end
-
-       it 'should not delete the manual decisions' do
-         expect(GreenFlag::FeatureDecision.find_by_id(@manual_fd.id)).to be_present
-       end
-
-       it 'should not create a feature event' do
-         expect(GreenFlag::FeatureEvent.count).to eq 0
-       end
-     end
-
-     context 'when there are automatic decisions' do
-       before(:each) do
-         @feature2 = GreenFlag::Feature.create!(code: 'huge_banner_ad')
-         @auto_fd = GreenFlag::FeatureDecision.create!(feature_id: @feature2.id, manual: false, enabled: true, site_visitor_id: 1)
-
-         @feature2.forget_non_manual_decisions!(enabled)
-       end
-
-       it 'should delete the automatic decisions' do
-         @feature2.forget_non_manual_decisions!(enabled)
-         expect(GreenFlag::FeatureDecision.find_by_id(@auto_fd)).to be_nil
-       end
-
-       it 'should create a feature event with the correct attributes' do
-         feature_event = GreenFlag::FeatureEvent.last
-         expect(feature_event).to be_present
-         expect(feature_event.feature_id).to eq @feature2.id
-         expect(feature_event.event_type_code).to eq GreenFlag::FeatureEvent::ENABLED_DECISIONS_FORGOTTEN
-         expect(feature_event.count).to eq 1
-       end
-     end
-   end
-
-   describe 'latest_version' do
-     let(:feature) { FactoryGirl.create(:green_flag_feature, version_number: 1) }
-
-     subject { feature.latest_version }
-
-     context 'when there are rules' do
-       before(:each) do
-         @latest_version_number = 15
-
-         (0...3).each do |i|
-           FactoryGirl.create(:green_flag_rule, feature_id: feature.id, version_number: @latest_version_number - i) 
-         end
-       end
-
-       it 'returns the version number of the last rule' do
-         expect(subject).to eq @latest_version_number
-       end
-     end
-
-     context 'when there are no rules' do
+      it 'returns the version number of the last rule' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.